File: services/db-admin/healthcheck.py
import http.client

def check_health():
    conn = http.client.HTTPConnection("localhost", 80, timeout=10)
    try:
        conn.request("GET", "/")
        response = conn.getresponse()
        return response.status == 200 or response.status == 302
    except Exception as e:
        print(f"Failed to connect: {e}")
        return False
    finally:
        conn.close()

if __name__ == "__main__":
    if check_health():
        print("Healthcheck passed")
        exit(0)
    else:
        print("Healthcheck failed")
        exit(1)

File: apps/cms/healthcheck.js
const http = require('http');

const options = {
  host: 'localhost',
  port: 1337,
  timeout: 2000,
};

const request = http.request(options, (res) => {
  if (res.statusCode === 200) {
    console.log('Healthcheck passed!');
    process.exit(0);
  } else {
    console.log('Healthcheck failed!');
    process.exit(1);
  }
});

request.on('error', function (err) {
  console.log('Healthcheck failed!', err);
  process.exit(1);
});

request.end();
